py3status/modules/coin_market.py

```diff
@@ -0,0 +1,223 @@
+        Please limit requests to no more than 10 per minute.
+        (default 600)
+    format: display format for this module (default '{format_coin}')
+    format_coin: display format for coins
+        (default '[\?color=#ff0 {name}] ${price_usd:.2f} {percent_change_24h}')
+    format_separator: show separator only if more than one (default ' ')
+    markets: number of top ranked markets or list of supported markets
+        (default ['btc'])
+    request_timeout: time to wait for a response, in seconds (default 5)
+    thresholds: thresholds to use for color changes
+        (default [(-100, 'bad'), (0, 'good')])
+
+Format placeholder:
+    {format_coin} format for cryptocurrency coins
+
+format_coin placeholders:
+    {24h_volume_usd}      eg 1435150000.0
+    {available_supply}    eg 16404825.0
+    {id}                  eg bitcoin
+    {last_updated}        eg 1498135152
+    {market_cap_usd}      eg 44119956596.0
+    {name}                eg Bitcoin
+    {percent_change_1h}   eg -0.17%
+    {percent_change_24h}  eg -1.93%
+    {percent_change_7d}   eg +14.73%
+    {price_btc}           eg 1.0
+    {price_usd}           eg 2689.45
+    {rank}                eg 1
+    {symbol}              eg BTC
+    {total_supply}        eg 16404825.0
+
+    Placeholders are retrieved directly from the URL.
+    The list was harvested only once and should not represent a full list.
+
+    To print coins in different currency, replace or replicate the placeholders
+    below with a valid option (eg '{price_gbp}') to create additional placeholders:
+
+    {price_xxx}           eg 1819.67786724  # GBP
+    {24h_volume_xxx}      eg 668740360.049  # GBP
+    {market_cap_xxx}      eg 29926922616.0  # GBP
+
+    Valid options are: AUD, BRL, CAD, CHF, CNY, EUR, GBP, HKD, IDR, INR, JPY,
+    KRW, MXN, RUB, otherwise USD. Valid options must be written in lowercase.
+
+Examples:
+```
+# view coins in GBP and optionally USD
+coin_market {
+    format_coin = '{name} £{price_gbp:.2f} ${price_usd:.2f} {percent_change_24h}'
+}
+```
+
+@author lasers, x86kernel
+
+SAMPLE OUTPUT
+[
+    {'color': '#FFFF00', 'full_text': 'Bitcoin '},
+    {'color': '#FFFFFF', 'full_text': '$2735.77 '},
+    {'color': '#00FF00', 'full_text': '+2.27%'},
+]
+
+losers
+[
+    {'color': '#FFFF00', 'full_text': 'Bitcoin '},
+    {'color': '#FFFFFF', 'full_text': '$2701.70 '},
+    {'color': '#FF0000', 'full_text': '-0.42%'},
+]
+"""
+
+
+class Py3status:
+    cache_timeout = 600
+    format = '{format_coin}'
+    format_coin = '[\?color=#ff0 {name}] ${price_usd:.2f} {percent_change_24h}'
+    format_separator = ' '
+    markets = ['btc']
+    request_timeout = 5
+    thresholds = [(-100, 'bad'), (0, 'good')]
+
+    def post_config_hook(self):
+        self.first_run = True
+        self.first_use = True
+        self.convert = None
+        self.url = self.reset_url = 'https://api.coinmarketcap.com/v1/ticker/'
+        self.limit = None
+
+        # ranked markets or cherrypicked markets?
```

Comment:
this comment asks a question rather than clarifying things.  Maybe I'm just confused here.

To be honest I don't like this overloading of parameters you are doing.